### ISSUE TITLE
InterfaceParserHelper Infinite Loop on Certain Patterns

### ISSUE THREAD

**Issue by JasonGore** [OfficeDev/office-ui-fabric-react#5528]
### Bug Report 
 
- __Package version(s)__: @uifabric/example-app-base 6.2.1
- __Browser and OS versions__: Chrome, Windows 10
 
#### Priorities and help requested:
 
Are you willing to submit a PR to fix? Yes
 
Requested priority: (Blocking, High, Normal, Low) Normal
 
Products/sites affected: apps/fabric-website-resources
 
#### Describe the issue:
 
InterfaceParserHelper can loop forever when certain patterns appear in source code. This causes Fabric website to loop forever until there is an out of memory crash when these patterns are present in source code loaded by the webpage.

This is an example of a pattern that triggers this behavior:

```
// export interface blah {
// }
export type ITextFieldStyleProps = number;
```

Debugging InterfaceParserHelper seems to show that it never parses past this block, looping forever until there is an OOM crash.

#### Actual behavior:

InterfaceParserHelper loops forever, preventing page from being loaded.
 
#### Expected behavior:
 
Don't loop forever

**Comment by cliffkoh** [comment]
@micahgodbolt Is there a way to tag low priority bugs?

**Comment by JasonGore** [comment]
I don't agree on low prioritization. We may never check in code like this, but it's very likely to happen in the course of development and affect people. I lost half a day to this issue thinking it was related to my code.

**Comment by micahgodbolt** [comment]
certainly isn't a pri0, so question is pri1 or pri2? I'd probably err on saying pri2 as it's not blocking anything (as annoying as it might be). Do we know the level of effort here?

**Comment by JasonGore** [comment]
Lower priority for sure. I tried fixing myself but didn't see an easy fix. It's possible someone who knows the code could fix it pretty easily.

**Comment by cliffkoh** [comment]
Given that this requires a very specific sequence, and has low occurrence (e.g. Micah, have you ever hit this before?), and will never be checked in, and doesn't affect production website, or affect any consumers of Fabric, it will be the very last bug I look at in my pile of bugs.

Hence low priority.

Do you disagree with this rationalization Micah? (e.g. should I look at this with equal priority to say #5162, possibly fixing this before #5162 since both are Priority 2s? (actually the other bug doesn't even have a priority - but it isn't 0 or 1))

**Comment by JasonGore** [comment]
The only concern I have is that the pattern is less specific than my example implies and the root cause is not obvious at all. People would not know to search issues for infinite loop, but maybe we can add keywords like "out of memory exception" and "hanging web browser" that will help them find this issue. Otherwise I agree for now that if nobody else has reported this issue we can assume it has low occurrence and low priority.

**Comment by cliffkoh** [comment]
Reopen since this is fixed in #5830.
